aten/src/ATen/function_wrapper.py

```diff
@@ -999,11 +1001,17 @@ def emit_body(env, option):
         elif ret['kind'] == 'type':
             assert len(calls) == 1
             call = calls[0]
-            if ret['type'] == 'THTensor*':
+            return_tensor_map = {
+                'THTensor*': '${Tensor}',
+                'THDenseTensor*': '${DenseTensor}',
+                'THDenseIndexTensor*': '${DenseBackend}LongTensor',
+            }
+            if ret['type'] in return_tensor_map.keys():
                 maybe_scalar = "->maybeScalar({})".format(scalar_check) \
                                if scalar_check is not None \
                                else ""
-                return_tensor = "return Tensor((new ${Tensor}(context,${arg_name}))${maybe_scalar},false);"
+                tensor_ctor = return_tensor_map[ret['type']]
+                return_tensor = "return Tensor((new %s(context,${arg_name}))${maybe_scalar},false);" % tensor_ctor
```

Comment:
Instead of mixing %s with the other formatting here, you can just add tensor_ctor to the list of substitutions.